front/src/modules/people/components/PeoplePicker.tsx

```diff
@@ -51,14 +64,20 @@ export function PeoplePicker({
     onSubmit(selectedPerson ?? null);
   }
 
+  let entitiesToSelect = people.entitiesToSelect;
+  if (company)
+    entitiesToSelect = people.entitiesToSelect.filter(
```

Comment:
this filtering should be integrated to useFilteredSearchEntityQuery logic. If not, we could end up with no result as people.entitiesToSelect only returns 10 people that could not be part of the desired company